src/main/java/commands/AddCommand.java

```diff
@@ -0,0 +1,40 @@
+package commands;
+
+import storage.Storage;
+import storage.TaskList;
+import ui.UI;
+
+/**
+ * Represents a command to add a new task to the task list.
+ * The AddCommand class implements the Command interface and handles the addition of a task.
+ */
+public class AddCommand implements Command {
+
+    private final String desc;
+
+    /**
+     * Constructs an AddCommand with the specified task description.
+     *
+     * @param desc the description of the task to be added
+     */
+    public AddCommand(String desc) {
+        this.desc = desc;
+    }
+
+    /**
+     * Executes the add task command, adding a new task to the task list.
+     * The task list is then saved to storage, and a line separator is printed.
+     *
+     * @param storage the Storage object for handling task persistence
+     * @param master the TaskList object containing the list of tasks
+     * @param ui the UI object for interacting with the user
+     * @return false, indicating that the application should not terminate
+     */
+    @Override
+    public boolean execute(Storage storage, TaskList master, UI ui) {
```

Comment:
The function was augmented to return a boolean function to distinguish between an exit command (which will return true) versus all other commands (which will return false), so that the program can catch and terminate when "bye" is inputted by the user. The aim is not to determine if command has execute successfully.

Noted on the Single Responsibility Principle. Let me try to think how I can improve this. Thank you!